src/descriptor.ts

```diff
@@ -2258,8 +2267,16 @@ export function processDescriptorRecursively(
   no_namespace: boolean,
   parentName: string = '',
 ): ts.Statement[] {
+  const objectTypeName = `ObjectTypeFor${parentName}${descriptor.name}`;
   const statements: ts.Statement[] = [
-    createMessage(rootDescriptor, descriptor, pbIdentifier, parentName),
+    ts.factory.createTypeAliasDeclaration(
+      undefined,
+      [ts.factory.createModifier(ts.SyntaxKind.ExportKeyword)],
+      ts.factory.createIdentifier(objectTypeName),
+      undefined,
+      createPrimitiveMessageSignature(rootDescriptor, descriptor),
+    ),
+    createMessage(rootDescriptor, descriptor, pbIdentifier, parentName, objectTypeName),
```

Comment:
The return type of `toObject` and the parameter of `fromObject` are supposed to be compatible.

Aside from preventing from duplicated source codes, we might wanna use only object type other than protocol Buffer. For example, [this open source project](https://github.com/pganalyze/libpg_query/tree/15-latest), which provides PostgreSQL parser, uses `.proto` file as a common language across several languages. So we actually need only data types corresponding to `.proto` file, instead of serializing/deserializing Protocol buffer and whatnot.